privacyidea/static/components/token/views/dialog.regenerate_token.html

```diff
@@ -0,0 +1,57 @@
+<div class="modal fade lock-dialog" id="dialogTokenRegenerate" tabindex="-1"
+         role="dialog"
+         ng-show="regenerateTokenStep < 5"
+         data-backdrop="static" data-keyboard="false"
+         aria-labelledby="myModalLabel"
+         aria-hidden="true">
+        <div class="modal-dialog modal-lg ">
+
+            <div class="modal-content" ng-show="regenerateTokenStep === 0">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span></button>
+                    <h4 class="modal-title" id="myModalLabel"
+                        translate>Regenerate Token</h4>
+                </div>
+                <div class="modal-body">
+                    <p translate ng-show="token.username">
+                    Do you really want to regenerate the Token {{ tokenSerial }} assigned to user {{ token.username }}?
+                    </p>
+                    <p translate ng-hide="token.username">
+                    Do you really want to regenerate the Token {{ tokenSerial }}?
+                    </p>
+                    <p translate>
+                        This will invalidate the old token and generate a new token secret.
+                    </p>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-default"
+                            data-dismiss="modal" translate>No</button>
+                    <button type="button"
+                            class="btn btn-danger"
+                            ng-click="nextRegenerateToken()">
+                        <span class="glyphicon glyphicon-refresh"></span>
+                        <span translate>Regenerate Token</span>
+                    </button>
+                </div>
+            </div>
+
+            <div class="modal-content" ng-show="regenerateTokenStep === 1">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span></button>
+                    <h4 class="modal-title" id="myModalLabel"
+                        translate>Token secret regenerated</h4>
+                </div>
+                <div class="modal-body">
+                    <ng-include src="instanceUrl + '/static/components/token/views/token.enrolled.' + selectedToken.tokentype + '.html'">
```

Comment:
We should only put the question, if the token should be rolled over in a modal dialog.
The rollover dialog itself should be a dedicated route.
This way, the rollover could also be addressed directly and we would follow the flat design.

To do so we need to add a route in static/components/token/states/states.js like this:

~~~~javascript
              .state('token.rollover', {
                    url: "/rollover/:tokenSerial",
                    templateUrl: tokenpath + "token.enrolled.hotp.html" + versioningSuffixProviderProvider.$get().$get(),
                    controller: "tokenDetailController",
                })
~~~~

However, we might need to add a wrapper html template so that we can handle different token types of the templates.

We could also put the javascript logic of rollover into its own controller.